ssl/ssl_cert.c

```diff
@@ -468,7 +469,8 @@ STACK_OF(X509_NAME) *SSL_dup_CA_list(const STACK_OF(X509_NAME) *sk)
         SSLerr(SSL_F_SSL_DUP_CA_LIST, ERR_R_MALLOC_FAILURE);
         return NULL;
     }
-    for (i = 0; i < sk_X509_NAME_num(sk); i++) {
+    sk_X509_NAME_reserve(ret, num);
+    for (i = 0; i < num; i++) {
         name = X509_NAME_dup(sk_X509_NAME_value(sk, i));
         if (name == NULL || !sk_X509_NAME_push(ret, name)) {
```

Comment:
If reserver guarantees that push can't fail for memory, why are we checking?  Is it this kind of thing that makes it a wip?

Reply:
The code either needs to check the return from the reserve call or from each individual insertion.  It doesn't need to do both.  I took the simpler (lazier?) route by not checking the reserve return code.  If the reserve fails, it is still possible the insertions might succeed.

Reply:
I'm actually with Rich on this one. In sense it would be more appropriate to check for return value from reserve call, rather than attempt to check individual insertions. Even though there is a chance that individual assertions can work (because another thread would free some memory after reservation attempt?), it's more likely to fail midway.